malloc_free/bpf/malloc_free.bpf.c

```diff
@@ -104,16 +130,26 @@ int BPF_KPROBE(uprobe_malloc, int size)
 	if (target_pid >= 0 && target_pid != pid)
 		return 0;
 
+	increment_stat(STAT_MALLOC_CALLS);
+
 	event->size = size;
 	event->tid = tid;
 	if (trace_path) {
+		u32 max_depth = max_stack_depth;
+		if (max_depth > PERF_MAX_STACK_DEPTH)
+			max_depth = PERF_MAX_STACK_DEPTH;
+		u32 stack_size = max_depth * sizeof(u64);
 		event->ustack_sz =
-			bpf_get_stack(ctx, event->ustack, sizeof(event->ustack),
+			bpf_get_stack(ctx, event->ustack, stack_size,
 				      BPF_F_USER_STACK);
 	} else {
 		event->ustack_sz = -1;
 	}
-	bpf_map_update_elem(&event_heap, &tid, event, BPF_NOEXIST);
+	
+	int ret = bpf_map_update_elem(&event_heap, &tid, event, BPF_NOEXIST);
+	if (ret != 0) {
+		increment_stat(STAT_EVENT_DROPS);
```

Comment:
Since `bpf_map_update_elem` might fails for various reasons, it is better to collect the statistics in more detail:

* `STAT_EVENT_DROPS_MAP_FULL`  for E2BIG : imply that user needs to increase max_entries
* `STAT_EVENT_DROPS_INVALID_KEY` for EEXIST and ENOENT: imply that there is an internal mismatch
* `STAT_EVENT_DROPS_NOMEM` for ENOMEM: imply out of memory happned
* `STAT_EVENT_DROPS_OTHERS' for other errors returned by `bpf_map_update_elem`: imply that there is an internal system error